cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/BasicMapReduceContext.java

```diff
@@ -237,6 +240,22 @@ private void addInput(String inputName, InputFormatProvider inputFormatProvider,
     if (mapperClass != null && !Mapper.class.isAssignableFrom(mapperClass)) {
       throw new IllegalArgumentException("Specified mapper class must extend Mapper.");
     }
+    // Record lineage only if the input format has a proper name
+    if (!inputName.equals(inputFormatProvider.getInputFormatClassName())) {
+      try {
+        // Create an external dataset for the input format for lineage tracking
```

Comment:
not sure I understand this. If I call addInput("one", fileSet1, null) and fileSet1 is a regular dataset (a FileSet), then you will create an externalDataset named "one-input"? So you create a "shadow" external dataset for every dataset is used as input in MR? Seems that something is wrong